pkg/client/unversioned/restclient_test.go

```diff
@@ -286,3 +289,39 @@ func TestDoRequestCreated(t *testing.T) {
 	}
 	fakeHandler.ValidateRequest(t, "/"+testapi.Default.Version()+"/test", "GET", nil)
 }
+
+func TestCreateBackoffManager(t *testing.T) {
+
+	theUrl, _ := url.Parse("http://localhost")
+
+	// 1 second base backoff + duration of 2 seconds -> exponential backoff for requests.
+	os.Setenv(envBackoffBase, "1")
+	os.Setenv(envBackoffDuration, "2")
```

Comment:
Id like to push back on this... unless you insist, because:

now that we've coupled the reading of env vars to the creation of the struct, we really need this test. 

**I agree** its an unwanted system side effect, but we need a good unit test for this functionality, and the cost of **not** having such a test is much higher than the cost of having a short lived environment var side effect.  

if you insist i will remove it but then i think maybe we should consider moving this into two separate functions to keep test coverage ratio.